reporters/GlobalSummaryReporter.js

```diff
@@ -105,6 +115,32 @@ class GlobalSummaryReporter {
 
     // Pricefeed stats.
     this.priceEstimate = this.referencePriceFeed.getCurrentPrice();
+
+    // Initialize Uniswap pair address to get trade data for. Default pair token is the collateral token.
+    this.uniswapPairToken = this.uniswapPairOverride[this.toChecksumAddress(this.empContract.options.address)]
+      ? this.uniswapPairOverride[this.toChecksumAddress(this.empContract.options.address)]
+      : this.collateralContract;
+    this.uniswapPairDetails = await getUniswapPairDetails(
+      this.web3,
+      this.syntheticContract.address,
+      this.uniswapPairToken.address
+    );
+    this.uniswapPairAddress = this.uniswapPairDetails.pairAddress.toLowerCase();
+
+    // Set up Uniswap subgraph client and query latest update stats.
+    // If the `latestSwapBlockNumber` < `endBlockNumberForPeriod`, then set `endBlockNumberForPeriod = latestSwapBlockNumber`
+    // otherwise the graphQL client will throw an error when trying to access a block that it has not indexed yet.
+    // Its ok to use the latest swap's block number as the end-period block number because we are only querying swap data for this pair.
+    // If we were using `endBlockNumberForPeriod` to query data for some other pair as well,
+    // then we wouldn't be able to set `endBlockNumberForPeriod = latestSwapBlockNumber`
+    this.uniswapClient = getUniswapClient();
+    const latestSwap = (await this.uniswapClient.request(queries.LAST_TRADE_FOR_PAIR(this.uniswapPairAddress))).pairs[0]
+      .swaps[0].transaction;
+    this.latestSwapTimestamp = latestSwap.timestamp;
+    this.latestSwapBlockNumber = Number(latestSwap.blockNumber);
+    if (this.endBlockNumberForPeriod > this.latestSwapBlockNumber) {
```

Comment:
New feature: grab latest swap block number and make sure we never query for blocks > this last swap block.